src/StripTags.php

```diff
@@ -179,19 +179,18 @@ public function filter($value)
         $value = (string) $value;
 
         // Strip HTML comments first
-        while (strpos($value, '<!--') !== false) {
-            $pos   = strrpos($value, '<!--');
-            $start = substr($value, 0, $pos);
-            $value = substr($value, $pos);
-
-            // If there is no comment closing tag, strip whole text
-            if (! preg_match('/--\s*>/s', $value)) {
-                $value = '';
+        $open     = '<!--';
+        $openLen  = strlen($open);
+        $close    = '-->';
```

Comment:
Not sure why, but before we expect closing tag to match reg exp `--\s*>`.
Now, we explicitly expecting no any spaces between `--` and `>`.

Do you think expecting some spaces there was a bug? I can't see any other test which fail because of that change, and also I haven't seen anything about these spaces in the specs.

Reply:
I couldn't see why it was doing that either. Running some additional tests this loops how handles comments in the same way `strip_tags` does. So i can only think this was a hidden feature?
With the string `test<!---- -- > --  > -->` the previous version returned `test --   --` where as now it returns `test`

Should the previous behaviour be restored?

Reply:
Honestly, I think the current behaviour is correct and comply with the html comment spec. Also, as you said, the same way `strip_tags` behaves, so I think it's right. You can add this test case as well.

Reply:
OK, cool. Test added